scripts/vagrant-linux-priv-config.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+export DEBIAN_FRONTEND=noninteractive
+
+# Update and ensure we have apt-add-repository
+apt-get update
+apt-get install -y software-properties-common
+
+# Add i386 architecture (for libraries)
+dpkg --add-architecture i386
+
+# Add a Golang PPA
+sudo add-apt-repository ppa:gophers/archive
+
+# Add the Docker repository
+apt-key adv \
+	--keyserver hkp://p80.pool.sks-keyservers.net:80 \
+	--recv-keys 9DC858229FC7DD38854AE2D88D81803C0EBFCD88
+add-apt-repository \
+	"deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+	$(lsb_release -cs) \
+	stable"
+
+# Update with i386, Go and Docker
+apt-get update
+
+# Install Core build utilities for Linux
+apt-get install -y \
+	build-essential \
+	git-core \
+	golang-1.9 \
+	libc6-dev-i386 \
+	liblxc1 \
+	libpcre3-dev \
+	linux-libc-dev:i386 \
+	lxc-dev \
+	lxc-templates \
+	pkg-config \
+	zip
+
+# Install Development utilities
+apt-get install -y \
+	curl \
+	default-jre \
+	docker-ce \
+	htop \
+	jq \
+	qemu \
+	silversearcher-ag \
+	tree \
+	unzip \
+	vim
+
+# Install ARM build utilities
+apt-get install -y \
+	binutils-aarch64-linux-gnu \
+	binutils-arm-linux-gnueabihf \
+	gcc-5-aarch64-linux-gnu \
+	gcc-5-arm-linux-gnueabihf \
+	gcc-5-multilib-arm-linux-gnueabihf
+
+# Install Windows build utilities
+apt-get install -y \
+	binutils-mingw-w64 \
+	gcc-mingw-w64
+
+# Ensure everything is up to date
+apt-get upgrade -y
+
+# Ensure Go is on PATH
+if [ ! -e /usr/bin/go ] ; then
+	ln -s /usr/lib/go-1.9/bin/go /usr/bin/go
```

Comment:
Please use a variable for the go version since we have to bump it with new releases.

I'd prefer using the official tarball, extracting it to `/usr/local/go`, and adding `/usr/local/go/bin` to `PATH` to avoid any subtle differences between upstream and the PPA. However, since it's built by an Ubuntu maintainer and appears fairly conservative in approach, we can try it out.

Reply:
As far as I can tell the package is fine, but the version can be extracted to a variable. It would be better if it added these links itself, obviously.